src/handler/mod.rs

```diff
@@ -766,15 +766,27 @@ impl Handler {
                 Ok((session, enr)) => {
                     // Receiving an AuthResponse must give us an up-to-date view of the node ENR.
                     // Verify the ENR is valid
-                    if self.verify_enr(&enr, &node_address) {
-                        // Session is valid
-                        // Notify the application
+                    if enr.node_id() == node_address.node_id {
+                        // Signature is valid, but socket may not be
```

Comment:
I didn't actually confirm in the source that the signature is validated before the node_id is set on `node_address`, I'm just hoping that's true. I could change this to simply:
```suggestion
                    if enr.node_id() == node_address.node_id {
                        // Node ID matches, but socket may not
```
Though at that point maybe I might as well remove the comment altogether...